File: objectiveai-rs/src/functions/inventions/response/unary/object.rs
use crate::functions::inventions::response;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
pub enum Object {
    #[serde(rename = "alpha.scalar.function.invention")]
    AlphaScalarFunctionInvention,
    #[serde(rename = "alpha.vector.function.invention")]
    AlphaVectorFunctionInvention,
}

impl From<response::streaming::Object> for Object {
    fn from(value: response::streaming::Object) -> Self {
        match value {
            response::streaming::Object::AlphaScalarFunctionInventionChunk => {
                Object::AlphaScalarFunctionInvention
            }
            response::streaming::Object::AlphaVectorFunctionInventionChunk => {
                Object::AlphaVectorFunctionInvention
            }
        }
    }
}

File: objectiveai-js/src/functions/inventions/response/unary/object.ts
import { z } from "zod";

export const FunctionsInventionsResponseUnaryObjectSchema = z.enum(["alpha.scalar.function.invention","alpha.vector.function.invention"]).meta({ title: "functions.inventions.response.unary.Object" });
export type FunctionsInventionsResponseUnaryObject = z.infer<typeof FunctionsInventionsResponseUnaryObjectSchema>;
